src/aks-preview/azext_aks_preview/_params.py

```diff
@@ -396,6 +397,8 @@ def load_arguments(self, _):
         c.argument('disk_driver_version', arg_type=get_enum_type(disk_driver_versions))
         c.argument('disable_disk_driver', action='store_true')
         c.argument('enable_file_driver', action='store_true')
+        c.argument('enable_namespace_resources', action='store_true', help='Enable sync of namespaces as Azure Resource Manager resources')
+        c.argument('disable_namespace_resources', action='store_true', help='Disable sync of namespaces as Azure Resource Manager resources')
```

Comment:
It would be better to add a mutual exclusion check for these two bool flags.

Reply:
@KarthikK123 Please take a look at this comment

Reply:
Added the validation in managed_cluster_decorator.py

Reply:
Cool, but you are missing unit test coverage for [line 1381](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=57567617&view=logs&j=1c778e8e-596b-5a0a-22c2-460364725d7c&t=3e82742e-7cc5-5f7d-8ec2-cd9417b56eb7&l=409) in `managed_cluster_decorator.py`.

Also, you may leverage the validator provided by cli framework. See the example of Microsoft Defender [params](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/_params.py#L354), [validator](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/_validators.py#L530-L532).

Reply:
Thanks for sharing the validator example, I changed it to use that pattern.